tron/core/actionrun.py

```diff
@@ -599,13 +601,16 @@ def _exit_unsuccessful(self, exit_status=None, retry_original_command=True) -> O
         if self.last_attempt is not None:
             self.last_attempt.exit(exit_status)
         if self.retries_remaining is not None:
-            if self.retries_remaining > 0:
-                self.retries_remaining -= 1
-                return self.restart(original_command=retry_original_command)
+            if exit_status in non_retryable_exit_codes:
+                log.info(f"{self} skipping auto-retries, received non-retryable exit code ({exit_status}).")
```

Comment:
@cuza think we should set `self.retries_remaining = 0` here as well? i was making sure that we didn't have a getter/property for retries_remaining that we used elsewhere and noticed that `retries_remaining` ends up getting stored in the serialized data that we use for persistence - i'm not sure what happens on a restart with that atm though

Reply:
Oh that's not a bad idea. I initially made `retries_remaining` but deleted cuz I thought it was overkill. I was not thinking about what would happen if we restart tho